OpenRA.Mods.Common/Traits/PaletteEffects/RotationPaletteEffect.cs

```diff
@@ -92,8 +92,8 @@ public void AdjustPalette(IReadOnlyDictionary<string, MutablePalette> palettes)
 
 			foreach (var kvp in palettes)
 			{
-				if ((info.Palettes.Count > 0 && !info.Palettes.Any(kvp.Key.StartsWith))
-					|| (info.ExcludePalettes.Count > 0 && info.ExcludePalettes.Any(kvp.Key.StartsWith)))
+				if ((info.Palettes.Count > 0 && !AnyPaletteNameStartsWith(info.Palettes, kvp.Key))
```

Comment:
In a more general vein - identifying things that are immutable on game start and making them actually immutable in the code is probably a good way to spot and eek out some small optimizations like this.